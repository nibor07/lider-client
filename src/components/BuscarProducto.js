import React, { useState } from 'react'
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const ContenedorImg = styled.img`
    width: 152px; 
    height: 35px; 
    margin-top: 12px; 
    margin-bottom: 6px;
`;

const ContenedorFondo = styled.div`
    background-color: rgb(0, 113, 206);
    font-size: 8pt;
    color: #fff;
`;

export const BuscarProducto = ( { setProductos } ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) =>{
        e.preventDefault();

        if( inputValue.trim().length > 2 ){
            setProductos(  prods => [ inputValue ]);
            setInputValue('');
        }
        
    }

    return (
            <ContenedorFondo className="col-xs-8">
            <div className="logo">
                <a href="/catalogo/">                    
                    <ContenedorImg alt="Lider Logo" src="https://www.lider.cl/catalogo/images/lider-logo.svg"/>
                </a>
            </div>        
            <div>
            <form className="col-12" onSubmit={ handleSubmit }>
            <div className="row mt-4">
                <div className="col-md-4">
                    <input
                        name="search"
                        value={ inputValue }
                        className="form-control"
                        type="text"
                        placeholder="Buscar por Producto"
                        onChange={handleInputChange}
                    />
                </div>

            </div>
        </form>
        </div>
        </ContenedorFondo>            
    )
}

BuscarProducto.propTypes = {
    setProductos: PropTypes.func.isRequired
}