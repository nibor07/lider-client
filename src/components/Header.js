import React, { Fragment } from 'react';
import styled from '@emotion/styled';

const ContenedorFondo = styled.div`
    background-color: #041e42;
    font-size: 8pt;
    color: #fff;
`;


const Header = () => {
    return(
        <Fragment>
        <ContenedorFondo>
        <span>Servicio al cliente de Lider.cl: WhatsApp 
            <a href="https://api.whatsapp.com/send?phone=56957211492&amp;text=Hola!%20Tengo%20una%20consulta">+56957211492
            </a>
             | Horario: de lunes a viernes de 8:00 a 22:00 y s√°bado, domingo y festivos de 9:00 a 22:00
        </span>
        </ContenedorFondo>
        </Fragment>
    );
}

export default Header;