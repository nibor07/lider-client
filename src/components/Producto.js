import React from 'react'

export const Producto = ({ producto }) => {

    let imagen = "https://"+producto.url;

    return (        
        <div className="col-md-4 mb-5">
            <div className="card">                 

                 <img className="card-img-top" src={imagen} alt={imagen} />
                 <h1 className="card-header" >{producto.brand} <span className="description"> {producto.description}</span></h1>
                 <p> { producto.price } </p>
                 <div className="card-body">
                     <button
                        type="button"
                        className="btn btn-block btn-light"
                        onClick={() => {
                        }}
                     >
                         Agregar
                     </button>

                 </div>
            </div>
        </div>

    )
}