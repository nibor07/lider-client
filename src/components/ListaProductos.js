import React from 'react';
import { useFetchProductos } from '../hooks/useFetchProductos';
import { Producto } from './Producto';

export const ListaProductos = ({ producto }) => {

    const { data:productos, loading } = useFetchProductos( producto );
    console.log('data=>', productos);
    if(productos){
        return (
            <>
                <h3 className="animate__animated animate__fadeIn"> { producto } </h3>

                { loading && <p className="animate__animated animate__flash">Loading</p> }

                <div className="card-grid">
                    
                    {
                        productos.map( producto => (
                            <Producto
                                key={ producto.id }
                                producto={producto}
                            />
                        ))
                    }
                
                </div>
            </>
        );
    }else{
        return(
            <div className="styled__NoResultsText-sc-1w20d9o-5 dTOjoM">
            No encontramos resultados para tu b√∫squeda de <span>{producto} </span>
        </div>
        )

    }
}