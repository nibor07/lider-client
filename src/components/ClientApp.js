import React, { useState } from 'react'
import { BuscarProducto } from './BuscarProducto';
import Header from './Header';
import { ListaProductos } from './ListaProductos';

export const ClientApp = () => {

    const [productos, setProductos] = useState([]);

    console.log( 'URL: ', process.env.REACT_APP_BACKEND_URL);
    return (
        <div>
            <Header />
            <BuscarProducto setProductos={ setProductos }/>
            <hr />
            <ol>
            {
                    productos.map( producto  => (
                        <ListaProductos 
                            key={ producto }
                            producto={ producto }
                        />
                    ))
            }
            </ol>     
        </div>
    )
}