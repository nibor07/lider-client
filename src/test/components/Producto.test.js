import React from 'react';
import { shallow } from 'enzyme'
import { Producto } from '../../components/Producto';

describe('Tests de <Producto />', () => {
    
    const producto = {  url: "wwww.test.cl/1.jpg",
                        brand: "Televisor DAEWOO",
                        price: "123"                        
                    };
    
    const wrapper = shallow( <Producto producto={ producto } /> );

    test('debe de mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de tener un pÃ¡rrafo con el brand del producto buscado', () => {
        
        const hdos = wrapper.find('p');
        expect( hdos.text().trim() ).toBe( producto.price );

    })

    *test('debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        // console.log( img.prop('src') )
        expect( img.prop('src') ).toBe( "https://"+producto.url );
        expect( img.prop('alt') ).toBe( "https://"+producto.url );

    });


    /*test('debe de tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__fadeIn') ).toBe( true );

    })*/
    
})
