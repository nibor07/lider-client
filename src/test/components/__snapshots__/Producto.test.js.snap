// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests de <Producto /> debe de mostrar el componente correctamente 1`] = `
<div
  className="col-md-4 mb-5"
>
  <div
    className="card"
  >
    <img
      alt="https://wwww.test.cl/1.jpg"
      className="card-img-top"
      src="https://wwww.test.cl/1.jpg"
    />
    <h1
      className="card-header"
    >
      Televisor DAEWOO
       
      <span
        className="description"
      >
         
      </span>
    </h1>
    <p>
       
      123
       
    </p>
    <div
      className="card-body"
    >
      <button
        className="btn btn-block btn-light"
        onClick={[Function]}
        type="button"
      >
        Agregar
      </button>
    </div>
  </div>
</div>
`;
